tests/org.dataflowanalysis.converter.tests/src/org/dataflowanalysis/converter/tests/PCMTest.java

```diff
@@ -87,68 +92,59 @@ private void testSpecificModel(String inputModel, String inputFile, String model
 
         var dfd = complete.dataFlowDiagram();
         var dd = complete.dataDictionary();
+        
 
-        assertEquals(dfd.getNodes()
-                .size(),
-                assIdToName.keySet()
-                        .size());
-
-        List<String> nodeIds = new ArrayList<>();
-        for (Node node : dfd.getNodes()) {
-            nodeIds.add(node.getId());
-        }
-        Collections.sort(nodeIds);
-        List<String> assIds = new ArrayList<>(assIdToName.keySet());
-        Collections.sort(assIds);
-
-        assertEquals(assIds, nodeIds);
```

Comment:
Please check that you "preserve" ids with your id,id_1,id_2 and so on logic

Reply:
Well.... We cannot preserve IDs from PCM -> DFD right now, as there will be multiple nodes created from the same PCM element. That should lead to collisions, because we cannot differentiate in some cases between the multiple instances of the DFD node.
This will always be the case, since we can call the same PCM SEFFs from multiple places with differing contexts. DFD Nodes do not have an attached context and thus cannot differentiate between those calls

Reply:
However, let's discuss this today

Reply:
Check the IDs before the `_`

Reply:
Implemented in 6acdf117b40c